sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/package.scala

```diff
@@ -204,6 +204,27 @@ package object util extends Logging {
     truncatedString(seq, "", sep, "", maxFields)
   }
 
+  /** Whether we have warned about plan string truncation yet. */
+  private val planSizeWarningPrinted = new AtomicBoolean(false)
+
+  def withSizeLimitedWriter[T](writer: Writer)(f: (Writer) => T): Option[T] = {
```

Comment:
Where is this being used? It is not being used for any of the actual plan logging right?